#115584041 Search Auto-complete

andela-joyebanji opened the pull request:
#### What does this PR do?

Enhance restaurant search with Auto-complete
#### Description of Task to be completed?
- Add route, `/restaurant/search`, in `route.php`
- Add controller action, `search`, in `RestaurantController.php` to handle search
- Include twitter's `typeahead` javascript library in `master.blade.php` to handle client-side auto-complete
- Add `typehead` class to search input box in `master.blade.php` and `search.blade.php`
- Add `typeahead-auto-complete.css` for styling search suggestions
#### How should this be manually tested?

> Note: Your search input must be greater that 1 character for you so get search suggestions

Landing page > input search text
Landing page > Click on any link on the site > type input search text on the compact search box at the top navigation 
#### What are the relevant pivotal tracker stories?

The corresponding story on `PT` can be viewed [here](https://www.pivotaltracker.com/story/show/115584041)

![screen shot 2016-04-12 at 2 56 19 pm](https://cloud.githubusercontent.com/assets/17027572/14462410/db6e1b1c-00be-11e6-8633-ac0bc13fed94.png)
![screen shot 2016-04-12 at 2 56 41 pm](https://cloud.githubusercontent.com/assets/17027572/14462411/db6e7634-00be-11e6-8e56-7b3e027d9819.png)

Commit:
[Feature #115584041] Search Auto-complete

- Add route, `/restaurant/search`, in `route.php`
- Add controller action, `search`, in `RestaurantController.php` to handle search
- Include twitter's `typeahead` javascript library in `master.blade.php` to handle client-side auto-complete
- Add `typehead` class to search input box in `master.blade.php` and `search.blade.php`
- Add `typeahead-auto-complete.css` for styling search suggestions

Please enter the commit message for your changes. Lines starting

andela-anandaa commented on public/css/typeahead-auto-complete.css:
```diff
@@ -0,0 +1,61 @@
+.twitter-typeahead{
```
There should be a space before `{` and after `:`.

andela-anandaa commented on public/css/typeahead-auto-complete.css:
```diff
@@ -0,0 +1,61 @@
+  -webkit-background-clip: padding-box;
+  -moz-background-clip: padding;
+  background-clip: padding-box;
+  width:102.1%;        
```
Same to this.

andela-anandaa commented on public/js/search_autocomplete.js:
```diff
@@ -0,0 +1,24 @@
+      source: restaurants,
+    });
+    $('#query.typehead').bind('typeahead:select', function(ev, suggestion) {
+      window.location.href = "/search?query="+suggestion.name
```
There should be a space before and after `+`.

andela-anandaa commented on public/js/search_autocomplete.js:
```diff
@@ -0,0 +1,24 @@
+var restaurants = new Bloodhound({
```
Should give some comments about this `Bloodhound` class.

Commit:
[Chore #115584041] Fix Nitpick issues on PR

- Add space after `:` and before `{` in typeahead-auto-complete.css
- Give comments about `Bloodhound` class library
